torch/csrc/autograd/engine.cpp

```diff
@@ -511,5 +525,8 @@ void GraphTask::init_to_execute(Function& graph_root, const edge_list& outputs)
   }
 }
 
+bool GraphTask::can_checkpoint() {
+  return exec_info.empty();
+}
```

Comment:
Can you please implement this in-line in the struct? It's such a simple method that there's little point in keeping it away